package-res/resources/web/vizapi/DataTable.js

```diff
@@ -15,1094 +15,1220 @@
+  var type = this._getColumn(colIdx).type;
+  return type ? type.toLowerCase() : "string";
+};
 
-*/
+/**
+ * Gets the id of the data property stored in a column, given its index.
+ *
+ * @method getColumnId
+ * @param {Number} colIdx The column index (zero-based).
+ * @return {String} The column's data property id.
+ */
+pentaho.AbstractDataTable.prototype.getColumnId = function(colIdx) {
+  return this._getColumn(colIdx).id;
+};
 
-pentaho = typeof pentaho == "undefined" ? {} : pentaho;
+/**
+ * Gets the label of the data property stored in a column, given its index.
+ *
+ * @method getColumnLabel
+ * @param {Number} colIdx The column index (zero-based).
+ * @return {String} The column's data property label.
+ */
+pentaho.AbstractDataTable.prototype.getColumnLabel = function(colIdx) {
+  return this._getColumn(colIdx).label;
+};
 
-/****************************************************
-    pentaho.DataTable
-    A client-side table object. 
-****************************************************/
+/**
+ * Gets the value of a column property.
+ *
+ * @method getColumnProperty
+ * @param {Number} colIdx The column index (zero-based).
+ * @param {String} name  The name of the property.
+ * @return {any} The property value.
+ */
+pentaho.AbstractDataTable.prototype.getColumnProperty = function(colIdx, name) {
+  return this._getColumn(colIdx)[name];
+};
 
-/*
-    Constructor.
-    jsonTable:      A CDA JSON table object or a Google DataTable JSON object
-*/
-pentaho.DataTable = function( jsonTable ) {
-    this.jsonTable = jsonTable;
-    this.className = "pentaho.DataTable";
-    if( jsonTable.metadata ) {
-        // convert from CDA to DataTable
-        this.jsonTable = pentaho.DataTable.convertCdaToDataTable(jsonTable);
-    }
-}
+/**
+ * Sets the value of a column property.
+ *
+ * @method setColumnProperty
+ * @param {Number} colIdx The column index (zero-based).
+ * @param {String} name  The name of the property.
+ * @param {any}    value The value of the property.
+ * @chainable
+ */
+pentaho.AbstractDataTable.prototype.setColumnProperty = function(colIdx, name, value) {
+  this._getColumn(colIdx)[name] = value;
+  return this;
+};
 
-/*
-    convertCdaToDataTable
-    Converts a CDA JSON table object to a Google DataTable JSON table object
-    
-    Input format
-    {
-        metadata: [
-            { colName: 'col1', colType: 'STRING', colLabel: 'Column 1' },
-            { colName: 'col2', colType: 'NUMERIC', colLabel: 'Column 2' }
-        ],
-        resultset: [
-            [ 'Row1', 123 ],
-            [ 'Row2', 456 ]
-        ]
-    }
-    
-    Output format
-    {
-        cols: [
-            { id: 'col1', type: 'string', label: 'Column 1' },
-            { id: 'col2', type: 'number', label: 'Column 2' },
-        ],
-        rows: [
-            { c: [ {v: 'Row 1' }, {v: 123} ] },
-            { c: [ {v: 'Row 2' }, {v: 456} ] }
-        ]
+// TODO: Consider caching results when no key function is specified.
```

Comment:
Identifies a possible future enhancement to the implementation of the following method.